jujugui/static/gui/src/app/components/entity-details/entity-details.js

```diff
@@ -151,10 +151,17 @@ YUI.add('entity-details', function() {
       if (data.length > 0) {
         var data = data[0];
         var model = this.props.makeEntityModel(data);
-        this.setState({entityModel: model}, () => {
+        this.setState({
+          entityModel: model,
+          previousPageTitle: document.title
+        }, () => {
           this._changeActiveComponent('entity-details');
           this._getPlans();
         });
+        var modelEntity = model.toEntity();
+        document.title = `${modelEntity.displayName}
```

Comment:
It's a little odd that a random component can modify the documents title, it also makes it more difficult to test.

app.js should probably have a method to set the document.title which can then be passed around to various components.